detox/utils/report.js

```diff
@@ -184,7 +184,7 @@ function removeOldGeneratedReports() {
     ].forEach((file) => fse.removeSync(`${ARTIFACTS_DIR}/${file}`));
 }
 
-function writeJsonToFile(jsonObject, filename, dir) {
+async function writeJsonToFile(jsonObject, filename, dir) {
     fse.writeJson(`${dir}/${filename}`, jsonObject).
         then(() => console.log('Successfully written:', filename)).
         catch((err) => console.error(err));
```

Comment:
Not sure what you want to achieve with this, but shouldn't be working. Even if you make this function async, it will not await for the result of `fse.writeJson`, since that is in a `promise.then().catch()` pattern, which wouldn't stop execution.

You would have to rewrite this as:
```suggestion
async function writeJsonToFile(jsonObject, filename, dir) {
    try {
        await fse.writeJson(`${dir}/${filename}`, jsonObject);
        console.log('Successfully written:', filename);
    } catch (err) {
        console.error(err);
    }
```

Reply:
Noted Daniel.